### DISCUSSION TITLE
Diagnostics module

### DISCUSSION THREAD

**Discussion by xkrt** [apache/pekko#2721] [General]
Hi!
It would be nice to have https://github.com/akka/akka-diagnostics analogue for Pekko. Especially Thread Starvation Detector.

**Comment by pjfanning** [comment]
That code has never been Apache licensed so it is out of bounds for us to even look at the code.

In theory, if someone could write code that has similar functionality and could prove that they never looked at the Akka code, then that would be a useful module for Pekko.

Could start out as an independent project and be brought into Pekko later.

**Comment by Philippus** [comment]
If I'm not mistaken, v2.0.0 is now available: https://github.com/akka/akka-diagnostics/pull/37.

**Comment by pjfanning** [comment]
I forked https://github.com/pjfanning/pekko-diagnostics and have a PR that is hopefully close to being ready (1 broken test).
That PR switches the code to Pekko. The fork is from the akka-diagnostics v2.0.0 tag.

My idea is to do a release under my own groupId in the next few weeks.

We can then look at bring it into the Pekko org but it could take a number of additional weeks or months to get a release out. There are other priorities in the Pekko org like getting 2.0.0 work on the existing modules done.

**Comment by pjfanning** [comment]
I published my fork as a milestone.

https://repo1.maven.org/maven2/com/github/pjfanning/pekko-diagnostics_2.13/0.1.0-M1/pekko-diagnostics_2.13-0.1.0-M1.pom

We can eventually look into moving the https://github.com/pjfanning/pekko-diagnostics repo into Apache Pekko org and doing ASF badged releases.
